device/edge-sample/edge_sample_module.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var Protocol = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+var SharedAccessKeyAuthenticationProvider = require('azure-iot-device').SharedAccessKeyAuthenticationProvider;
+var fs = require('fs');
+
+var connectionString = process.env.EdgeHubConnectionString;
+
+var authProvider = SharedAccessKeyAuthenticationProvider.fromConnectionString(connectionString);
+authProvider.getDeviceCredentials(function(err, credentials) {
+  if (err) {
+    throw new Error('unexpected: getDeviceCredentials failure');
+  } else {
+    credentials.ca = fs.readFileSync(process.env.EdgeModuleCACertificateFile).toString('ascii');
```

Comment:
>credentials.ca  [](start = 4, length = 15)

I agree with your "skeevy" comment seeing this here. I think we have to figure out a way for the auth provider to return authentication for IoT Hub AND the edge hub. maybe with a completely new authentication provider type?